internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -337,6 +337,7 @@ func (f *FleetGateway) execute(ctx context.Context) (*fleetapi.CheckinResponse,
 	ecsMeta, err := info.Metadata(ctx, f.log)
 	if err != nil {
 		f.log.Error(errors.New("failed to load metadata", err))
+		return nil, 0, err
```

Comment:
I noticed we did this by looking at an SDH in 8.16 where we probably send empty or malformed metadata.

I think this approach is incorrect because it will prevent agent from checking in at all. This includes checking in to get upgrade actions to fix this bug. We need to not panic but degrade the content of the checkin payload instead.